### PR TITLE
Adds pagination to extension request list

### PR DESCRIPTION
**Pull request by Aryex82**
Issue: https://github.com/Real-Dev-Squad/website-dashboard/issues/430

### What is the change?
- Currently we are able to filter api data based on only one value for a query param eg:  `?q=status:APPROVED`. This PR allows us to filter the data based on multiple values eg: `?q=status:APPROVED+PENDING`
- This PR also adds pagination support for the api. To get paginated result we would have to pass the required page size `?size=10`.
- This PR also adds a functionality to sort the data based on `created time`.eg: `?order=asc`
- Adds a new utility function to parse [RDS query language](https://github.com/Real-Dev-Squad/website-backend/pull/1419#pullrequestreview-1581088590)
### Note to Reviewers:
- The feature is under a feature flag. To access, please use `?dev=true` query param.
- Please also review this PR which changes the query param format in FE: - https://github.com/Real-Dev-Squad/website-dashboard/pull/470
- Please also review this PR which changes the query param format in backend: - https://github.com/Real-Dev-Squad/website-backend/pull/1454
### API contract: 
https://github.com/Real-Dev-Squad/website-api-contracts/pull/154
### Test stats
Before : 
Screenshot 2023-08-15 at 8 15 17 PM

After: 
Screenshot 2023-08-15 at 8 16 11 PM

Screenshot 2023-08-24 at 7 53 54 PM

### \*Dev Tested?

- [x] Yes
- [ ] No

### Screenshot: 
Screenshot 2023-08-28 at 12 43 35 AM

### Indexes : 
The following indexes needs to be created with these change: 
Screenshot 2023-08-15 at 9 32 52 PM

### CODE REVIEW

**Review by RitikJaiswal75 — COMMENTED** [review]
hey Ajeyakrishna, In case we want to use multiple params we must use query params like
`q=status:assigned+available` we wont be using `&` (Moving to RDS's own query language)

**Comment by Aryex82** [comment]
> hey Ajeyakrishna, In case we want to use multiple params we must use query params like
> `q=status:assigned+available` we wont be using `&` (Moving to RDS's own query language)

I took inspiration for this design from an API that's currently being used in RDS: 
Screenshot 2023-08-17 at 7 08 51 AM


I don't mind making your requested changes for this PR. 

But I also think we should create a separate issue to identify and migrate the other API which are not using the format that you've mentioned.

Let me know your thoughts on the same.

**Comment by Aryex82** [comment]
> hey Ajeyakrishna, In case we want to use multiple params we must use query params like
> `q=status:assigned+available` we wont be using `&` (Moving to RDS's own query language)

@RitikJaiswal75 I need a more clarity on this. 
What if we have two keys with multiple values eg: `?status=APPROVED&status=PENDING&assignee=user1&assignee=user2`. 
How will we handle this in the format that you have mentioned

File: utils/extensionRequests.js [hunk 1]
```diff
@@ -25,7 +24,39 @@ const formatExtensionRequest = async (extensionRequest) => {
   return { ...body, id, timestamp, assignee };
 };
 
+const transformQuery = (dev = false, size) => {
```
Comment (apurvkhare) [inline utils/extensionRequests.js:27]: When possible, keep the default params last in the order of params.
Currently, someone can never call `transformQuery` by just passing the `size` argument.
But if you interchange the positions of the params then you can.

Comment (Aryex82) [reply utils/extensionRequests.js]: Made the necessary changes.

**Review by apurvkhare — APPROVED** [review]

**Comment by prakashchoudhary07** [comment]
Please update the API contracts for it

**Comment by Aryex82** [comment]
> Please update the API contracts for it

Ive raised a PR for the same: https://github.com/Real-Dev-Squad/website-api-contracts/pull/154

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**